### ISSUE TITLE
Video Doorbell Service

### ISSUE THREAD

**Issue by xxxchris** [dgreif/ring#111]
Hi. Is it possible to make a service that connects camera, the motion sensor or ring button, a light and a door lock?
I have seen that in other Homebridge Plugins. (For example Homebridge-homematic)
There you get the notification when a motion or button is pressed. A preview is also delivered of the cam and you can turn on light or open the door in that rich notification.

**Comment by dgreif** [comment]
I'm a little confused because what your asking for is already set up.  Have you tried using the homebridge plugin yet?  The cameras will combine the motion sensor, light (if equipped), doorbell button (if equipped) and camera into one device in HomeKit.  If you put that camera in the same room as a HomeKit lock in the Home app, you will get an `Unlock Door` shortcut in the notifications.

**Comment by xxxchris** [comment]
Yeah. It worked with the push of the button. I thought the same rich notification would work with the motion sensor. only problem with the unlock button, my lock is only unlocking but not pulling the snapper to open the door. I think it's another datapoint the lock mechanism should get. but I think at the moment it's a standard home kit combination. 
It is possible to create a dedicated special device which I can tell which things it should trigger in the rich notification and how. They are doing and explaining it here: https://github.com/thkl/homebridge-homematic/wiki/Video-DoorBell
I will look into it. Maybe I can combine it with the ring plugin you created.

BTW: great development. thanks a lot.  I ever read that it will not be possible to get the SIP video communication in a home kit compatible format.

**Comment by dgreif** [comment]
@xxxchris are the lock and the "snapper" two different devices in HomeKit?  If so, you could try setting up an automation that would trigger the snapper whenever the lock is unlocked.

**Comment by xxxchris** [comment]
It‘s all in one device from a system called Homematic.
I don’t know how the locks work in the US. But in Germany most of the doors are technically like that. https://youtu.be/LDoVF9itiTA 
The snapper (I don’t know the correct english word, I even don‘t know if it‘s the right word in german) is the thing which you pull back when you turn the key a half round more as the unlock position of the key and then the door is open. 
So I have a lock icon in homekit which I can unlock and lock. And also a parameter which means turn the key a half round more. It’s just configurable in the eve app. Because it presents more paramaters but is still a homekit characteristic and not any special eve device. 
The rich notification uses only the standard parameter of the lock icon. The door only unlocks and not opens.

**Comment by xxxchris** [comment]
I saw yesterday that the motion sensor also delivers a picture of the cam in the notification. But not every time. Don’t know why the behavior is like this. 
For the lock maybe I can implement a second lock with a different behavior that opens and not only unlocks. I think that is the solution.

**Comment by xxxchris** [comment]
Or I‘ll wait for iOS 13. 
just found that https://data.mactechnews.de/543754.jpg

**Comment by xxxchris** [comment]
Maybe you've got a solution for the motion sensor, which only sometimes deliver a picture in the notification. it seems that if I've opened up the home app before a motion is detected a picture of the camera will be delivered. but when it did not have an actual picture it doesn't deliver a picture. maybe it depends on the actuality of the picture if it is delivered with the notification or not? so if you trigger getting the picture when the motion ist detected maybe it is delivered within the notification.

**Comment by xxxchris** [comment]
Yeah. I thought that. Maybe better I switch the to the Netatmo Doorbell. When it is released. Did you hear about that allready?
